src/actions/index.js

```diff
@@ -45,7 +45,8 @@ export const submitApp = form => (dispatch) => {
   const user = firebase.auth().currentUser;
 
   if (user) {
-    applicationsRef.doc(user.uid).set(form);
+    const newForm = { ...form, time: Date.now() };
```

Comment:
This is vulnerable to manipulation of the system's time. Using the server's time is better.
See: https://firebase.google.com/docs/reference/js/firebase.firestore.Timestamp

```suggestion
    const newForm = { ...form, time: firebase.firestore.Timestamp.now() };
```